### PR TITLE
Minor Fixes

### PR DESCRIPTION
**Pull request by gandarez**
1. Bug fix when deleting proxy configuration from SettingsForm. It's related to #29.
2. Accordingly to issue #30 I've changed the way of logging.
image

### CODE REVIEW

File: Downloader.cs [hunk 1]
```diff
@@ -8,35 +8,23 @@ class Downloader
     {
         static public void DownloadCli(string url, string dir)
         {
-            Logger.Instance.Info("Downloading wakatime cli...");
```
Comment (alanhamlett) [inline Downloader.cs]: We need this log line here, but it should be `Logger.Debug` instead of `Logger.Instance.Info`.

File: Logger.cs [hunk 2]
```diff
@@ -1,39 +1,58 @@
...
+            var outputPaneGuid = new Guid(GuidList.GuidWakatimeOutputPane.ToByteArray());
+            IVsOutputWindowPane windowPane;
+
+            outputWindow.CreatePane(ref outputPaneGuid, "Wakatime", 1, 1);
+            outputWindow.GetPane(ref outputPaneGuid, out windowPane);
+
+            return windowPane;
+        }
+        
+        internal static void ExceptionWriteLine(string message, Exception ex = null)
```
Comment (alanhamlett) [inline Logger.cs]: Let's rename this to `Error`.

File: Logger.cs [hunk 3]
```diff
@@ -1,39 +1,58 @@
...
+        {
+            var exceptionMessage = string.Format("{0}: {1}", message, ex);
+
+            WriteLine("Handled Exception", exceptionMessage);
         }
 
-        public void Error(string message)
+        internal static void WarningWriteLine(string message)
```
Comment (alanhamlett) [inline Logger.cs]: Let's rename this to `Warning`.

File: Logger.cs [hunk 4]
```diff
@@ -1,39 +1,58 @@
...
         {
-            _log.LogEntry((UInt32)__ACTIVITYLOG_ENTRYTYPE.ALE_ERROR,
-                ToString(),
-                string.Format(CultureInfo.CurrentCulture,
-                    "{0}", message));
+            WriteLine("Warning", message);
         }
 
-        public void Info(string message)
+        internal static void InfoWriteLine(string message)
```
Comment (alanhamlett) [inline Logger.cs]: Let's rename this to `Info`.

**Comment by gandarez** [comment]
@alanhamlett refactoring done!

File: WakaTimePackage.cs [hunk 5]
```diff
@@ -221,25 +220,14 @@ public static void SendHeartbeat(string fileName, bool isWrite)
             var pythonBinary = PythonManager.GetPython();
             if (pythonBinary != null)
             {
-
                 var process = new RunProcess(pythonBinary, arguments.ToArray());
-                if (DEBUG)
-                {
-                    Logger.Instance.Info("[\"" + pythonBinary + "\", \"" + string.Join("\", ", arguments) + "\"]");
```
Comment (alanhamlett) [inline WakaTimePackage.cs]: Let's add these log lines back with the new `Logger.Debug` method. This will greatly help debugging any future problems.

**Comment by gandarez** [comment]
@alanhamlett done! 
I also improved a little bit when saving settings.

File: PythonManager.cs [hunk 6]
```diff
@@ -44,12 +44,7 @@ static string GetPathFromMicrosoftRegister()
 
                 process.Run();
 
-                if (!process.Success)
```
Comment (alanhamlett) [inline PythonManager.cs]: I prefer the if condition broken out, and a `Logger.Debug` line showing which method was used to find Python.

Comment (gandarez) [inline PythonManager.cs]: I also follow the suggestion made by Resharper to reduce nesting.
http://stackoverflow.com/questions/268132/invert-if-statement-to-reduce-nesting and https://confluence.jetbrains.com/display/ReSharper/Invert+'if'+to+reduce+nesting

**Comment by gandarez** [comment]
@alanhamlett added verbose to PythonManager :+1:

File: PythonManager.cs [hunk 7]
```diff
@@ -44,17 +44,17 @@ static string GetPathFromMicrosoftRegister()
...
             }
-            catch (Exception)
+            catch (Exception ex)
             {
+                Logger.Error("GetPathFromMicrosoftRegister:", ex);
                 return null;
             }
+            finally
+            {
+                Logger.Debug("Python found by Microsoft Register");
```
Comment (alanhamlett) [inline PythonManager.cs]: Will this code be reached if there is a return statement above it?

**Comment by alanhamlett** [comment]
Merged it!

File: PythonManager.cs (hunk 7 above)
Comment (gandarez) [inline PythonManager.cs]: Yes, always. Finally block is implemented exactly to do it even if there's a previous return statement.

**Pull request merged**